lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -172,24 +174,24 @@ def _masker(item):
             result = result.data
     else:
         result = ma.masked
-        if value != _MDI and factor != _MDI:
+        if value not in _MDIs and factor not in _MDIs:
             result = _unscale(value, factor)
     return result
 
 
 # Regulations 92.1.4 and 92.1.5.
-_MDI = 2 ** 32 - 1
+_MDIs = [2 ** 31 - 1, 2 ** 32 - 1]
```

Comment:
The only real substantive change is this one, as far as I can see. Sorry for making you pull teeth, but are you able to talk me through why there are 2 MDIs?

The GRIB spec talks of:

**92.1.4**: All bits set to “1” for any value indicates that value is missing. This rule shall not apply to
packed data.

**92.1.5**: If applicable, negative values shall be indicated by setting the most significant bit to “1”

So I agree that for unsigned data the MDI is ```2 ** 32 - 1```, and for signed it is ```2 ** 31 - 1```, but ```2 ** 31 - 1``` most certainly isn't an MDI for signed. Treating it in this way cuts out the extra range that unsigned data gives and mis-represents data as MDI when it is not intended to be.